### PR TITLE
[octree] Replace usage of `std::vector<int>` with `Indices` in octree module & tests

### PR DESCRIPTION
**Pull request by haritha-j**
Instances of `std::size_t` ignored for now. 
Need to address requirement for returning negative indices.
Use of `uint32_t` for width ignored for now.

### CODE REVIEW

File: octree/include/pcl/octree/octree_search.h [hunk 1]
```diff
@@ -342,7 +342,7 @@ class OctreePointCloudSearch
      * \param[in] point_idx index for a dataset point given by \a setInputCloud
      * \param[in] point_distance distance of query point to voxel center
      */
-    prioPointQueueEntry(unsigned int& point_idx, float point_distance)
+    prioPointQueueEntry(index_t& point_idx, float point_distance)
```
Comment (kunaltyagi) [inline octree/include/pcl/octree/octree_search.h:345]: API breakage here

File: octree/include/pcl/octree/octree_pointcloud.h [hunk 2]
```diff
@@ -448,7 +449,7 @@ class OctreePointCloud : public OctreeT {
    * \return PointT from input pointcloud dataset
    */
   const PointT&
-  getPointByIndex(const unsigned int index_arg) const;
+  getPointByIndex(const index_t index_arg) const;
```
Comment (kunaltyagi) [inline octree/include/pcl/octree/octree_pointcloud.h:452]: API break here

File: octree/include/pcl/octree/impl/octree_pointcloud.hpp [hunk 3]
```diff
@@ -700,10 +700,10 @@ template <typename PointT,
           typename OctreeT>
 const PointT&
 pcl::octree::OctreePointCloud<PointT, LeafContainerT, BranchContainerT, OctreeT>::
-    getPointByIndex(const unsigned int index_arg) const
+    getPointByIndex(const index_t index_arg) const
```
Comment (kunaltyagi) [inline octree/include/pcl/octree/impl/octree_pointcloud.hpp:703]: API break

**Comment by kunaltyagi** [comment]
@haritha-j When you get back to this, please update the PR with the policy of adding deprecated functions to update the API, which has been used in other PRs

**Comment by mvieth** [comment]
I would say this pull request can be closed, because pull request #4350 has already made these changes. Any objections?

**Comment by kunaltyagi** [comment]
None from my side

**Pull request closed without merging**